api/src/license_operator.rs

```diff
@@ -0,0 +1,207 @@
+use std::borrow::Borrow;
+
+use log::info;
+use rand::rngs::ThreadRng;
+use rand::Rng;
+use sha3::{
+    digest::{ExtendableOutput, Update, XofReader},
+    Shake256,
+};
+
+use crate::adler32::adler32_checksum;
+use crate::license_key::LicenseKeyStatus;
+use crate::license_magic::LicenseMagic;
+use crate::license_serializer::{DefaultLicenseKeySerializer, LicenseKeySerializer};
+use crate::LicenseKey;
+
+pub struct LicenseOperator {
+    rng: ThreadRng,
+
+    magic: LicenseMagic,
+    key_size: usize,
+    checksum_init: (u32, u32),
+
+    serializer: Box<dyn LicenseKeySerializer>,
+
+    blacklist: Vec<Vec<u8>>,
+    byte_check: Vec<(usize, LicenseMagic)>,
```

Comment:
What is the `usize` element for?

Reply:
It is used for byte position which you want to be checked if it is valid with LicenseMagic.

For example, you want to check if third byte is valid from vector of bytes (serialized_key).